### PR TITLE
Rewrite `.feature` files to use the `"""` syntax

### LINKED ISSUE #8342: Adjust to official Gherkin parser in pytest-bdd
In its [latest pre-release](https://github.com/pytest-dev/pytest-bdd/blob/master/CHANGES.rst), `pytest-bdd` [switched to](https://github.com/pytest-dev/pytest-bdd/pull/698) using the [official Gherkin parser](https://github.com/cucumber/gherkin).

In-line with the [Gherkin docs](https://cucumber.io/docs/gherkin/reference/#step-arguments), that means multi-line steps are now unsupported (also see https://github.com/pytest-dev/pytest-bdd/issues/123 for a rationale).

That means that our files currently fail to parse, [breaking bleeding-edge CI](https://github.com/qutebrowser/qutebrowser/actions/runs/11338617331/job/31532124660):

```
[...]
E   gherkin.errors.CompositeParserException: Parser errors:
E   (16:13): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got 'windows:'
E   (17:13): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got '- tabs:'
E   (18:15): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got '- history:'
E   (19:17): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got '- url: http://localhost:*/data/backforward/1.txt'
E   (20:17): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got '- active: true'
E   (21:19): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got 'url: http://localhost:*/data/backforward/2.txt'
E   (32:13): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got 'windows:'
E   (33:13): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got '- tabs:'
E   (34:15): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got '- history:'
E   (35:17): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got '- url: http://localhost:*/data/backforward/1.txt'
E   (36:17): expected: #EOF, #TableRow, #DocStringSeparator, #StepLine, #TagLine, #ExamplesLine, #ScenarioLine, #RuleLine, #Comment, #Empty, got '- active: true'

The above exception was the direct cause of the following exception:
[...]
E   pytest_bdd.exceptions.TokenError: Unexpected token found. Check Gherkin syntax near the reported error.
E   Line number: 16
E   Line:             windows:
E   File: /__w/qutebrowser/qutebrowser/tests/end2end/features/backforward.feature
```

Based on the documentation, we will need to do either:

```gherkin
        Then the session should look like:
            """
            windows:
            - tabs:
              - active: true
                history:
                - active: true
                  url: http://localhost:*/data/backforward/1.txt
            """
```

or possibly

````gherkin
        Then the session should look like:
            ```yaml
            windows:
            - tabs:
              - active: true
                history:
                - active: true
                  url: http://localhost:*/data/backforward/1.txt
            ```
````

but only the former [is supported](https://github.com/pytest-dev/pytest-bdd/blob/7.3.0/src/pytest_bdd/parser.py#L329) by the current parser, and the docs also point out:

> Whilst all current versions of Cucumber support backticks as the delimiter, many tools like text editors don’t (yet).
> Whilst all current versions of Cucumber support content types as the delimiter, many tools like text editors don’t (yet).

Would be cool if e.g. VS Code had YAML syntax highlighting in there, but eh, not that important.

---

So it looks like we need to rewrite our `.feature` files to use the `"""` syntax (so they stay compatible before we upgrade, but still work after). Marking this as a "good first issue", as while it's some "boring" work, it'd help qutebrowser to stay up-to-date with pytest-bdd, and it's easy to validate whether things still work.

Here's a demo change:

```patch
diff --git i/tests/end2end/features/backforward.feature w/tests/end2end/features/backforward.feature
index d60fde645..93e9788d4 100644
--- i/tests/end2end/features/backforward.feature
+++ w/tests/end2end/features/backforward.feature
@@ -47,12 +47,14 @@ Feature: Going back and forward.
         And I run :back -t
         Then the error "At beginning of history." should be shown
         Then the session should look like:
+            """
             windows:
             - tabs:
               - active: true
                 history:
                 - active: true
                   url: http://localhost:*/data/backforward/1.txt
+            """
 
     Scenario: Going back in a new background tab
         Given I open data/backforward/1.txt
```

a similar change should be made for all multi-line steps in all feature files.

The end result should be that:

- `tox -e py312-pyqt67 -- tests/end2end/features` runs successfully (adjust Python version as needed)<br>(potentially tests will fail due to environment differences - if so, try without any changes at all, and if they still fail, it's probably okay)
- After doing `.tox/py312-pyqt67/bin/pip install -U --pre pytest-bdd>=8.0.0b1` (again, adjust Python version as needed), the tests should still run.
- You can downgrade again by running the same command, but with `<8`

### PR DESCRIPTION
**Pull request by mschilli87**
fixes #8342

This is untested and purely based on the information provided in the referenced issue.
But it looked pretty straightforward to me so I just used a bit of `git grep` + `vim`-foo to insert the `"""` where I understood they would be required now, adjusting the indentation appropriately.

Let me know if I missed anything or messed up in any other way.

### CODE REVIEW

**Review by The-Compiler — CHANGES_REQUESTED** [review]

File: tests/end2end/features/backforward.feature [hunk 1]
```diff
@@ -13,12 +13,14 @@ Feature: Going back and forward.
         And I wait until data/backforward/2.txt is loaded
         And I reload
         Then the session should look like:
+	    """
```
[the remark this answers is not in the document]
Comment (The-Compiler) [reply tests/end2end/features/backforward.feature:16]: Huh, funny, GitHub shows it as indented in the comments view, but not in the "files changed" view. I think this is mixed indentation with `\t` plus four spaces now, which it certainly shouldn't be.

[the remark this answers is not in the document]
Comment (mschilli87) [reply tests/end2end/features/backforward.feature:16]: Yes, looks like my vim-settings messed that up. 😅 What indentation do you prefer? Should be an easy fix.

[the remark this answers is not in the document]
Comment (The-Compiler) [reply tests/end2end/features/backforward.feature:16]: We mostly use four spaces in those files I believe. Could be that it's inconsistent unfortunately (I think some places there might use two spaces?).

**Comment by The-Compiler** [comment]
Thanks for getting the ball rolling! Done with a couple of more changes:

- 629a38d65 Upgrade to pytest-bdd 8
- f52492621 Fix tab indentation in feature files
- bb462c260 pytest-bdd 8: Fix indentations
- cec00266d pytest-bdd 8: Fix tags
- a8a5068e6 pytest-bdd 8: Fix missing multiline step quoting
- f4260a6e5 pytest-bdd 8: Fix comments and remaining syntax issues
- f91dc2d52 pytest-bdd 8: Adjust Python code for multiline steps
- 59fd7876a pytest-bdd 8: Fix borked tag
- aed06c5f4 Add gherkin-official changelog URL
- e2fb1fba8 Revert "Bleeding requirements: Use pytest-bdd release"
- acc06cdd3 Merge branch 'gherkin8342'
- a7b566e7b Remove pytest-bdd<8 limit

**Pull request merged**